docs/proposals/cri.md

```diff
@@ -0,0 +1,313 @@
+---
+title: CRI Design for edged
+authors:
+    - "@gpinaik"
+approvers:
+  - "@qizha"
+  - "@CindyXing"
+  - "@Baoqiang-Zhang"
+  - "@m1093782566"
+creation-date: 2019-04-19
+last-updated: 2019-04-21
+status: alpha
+---
+
+# CRI Support in edged
+
+* [CRI Support in edged](#CRI-Support-in-edged)
+  * [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non\-goals](#non-goals)
+  * [Proposal](#proposal)
+    * [Use Cases](#use-cases)
+  * [High Level Design](#high-level-design)  
+    * [Edged with CRI support](#edged-with-cri-support)
+  * [Low Level Design](#low-level-design)  
+    * [Packages](#packages)
+    * [Configuration parameters](#configuration-parameters)
+    * [Data structure modifications](#data-structure-modifications)
+    * [Edged object creation modifications](#edged-object-creation-modifications)
+    * [Runtime dependent module modifications](#runtime-dependent-module-modifications)
+    * [Runtime dependent functional modifications](#runtime-dependent-functional-modifications)
+  * [Open Questions](#open-questions) 
+    
+    
+## Motivation
+This proposal addresses the Container Runtime Interface support in edged to enable supporting multiple container
+runtimes like docker, containerd, crio etc on the edge platform.
+
+### Goals
+CRI support in edged must:
+* support multiple runtimes like docker, contianerd, crio etc.
+* Support corresponding CNI with pause container and IP
+
+### Non-goals
+
+* Automatic detection of container runtimes and its selection.
+* 
+## Proposal
+
+Currently Kubernetes kubelet CRI supports container runtimes like containerd, crio etc but support for docker runtime is
+provided using dockershim as well. However going forward even docker runtime will be supported through only CRI. Also 
```

Comment:
going forward k8s community will be removing kubelet dependency from dockershim and use CRI

Reply:
Do we understand how community is going to change the dockershim and related? Any impact to our design?

Reply:
Check kubernetes issue 73933...assume that we will have a separate cri-dockerd which should work with the existing kubelet CRI...since we are trying to incorporate the same CRI support we should not have any impact on the design